File: zig/src/indicators/core/output_descriptor.zig
const shape = @import("outputs/shape.zig");
const role_mod = @import("role.zig");
const pane_mod = @import("pane.zig");

/// Classifies a single indicator output for charting / discovery.
pub const OutputDescriptor = struct {
    /// Integer representation of the output enumeration of the related indicator.
    kind: i32,
    /// The data shape of this output.
    shape: shape.Shape,
    /// The semantic role of this output.
    role: role_mod.Role,
    /// The chart pane on which this output is drawn.
    pane: pane_mod.Pane,
};

File: rs/src/indicators/core/output_descriptor.rs
use super::outputs::shape::Shape;
use super::pane::Pane;
use super::role::Role;

/// Classifies a single indicator output for charting / discovery.
#[derive(Debug, Clone)]
pub struct OutputDescriptor {
    /// Integer representation of the output enumeration of the related indicator.
    pub kind: i32,
    /// The data shape of this output.
    pub shape: Shape,
    /// The semantic role of this output.
    pub role: Role,
    /// The chart pane on which this output is drawn.
    pub pane: Pane,
}
